test/e2e/util/util.go

```diff
@@ -216,6 +214,92 @@ func WaitForNUnReadyNodes(t *testing.T, ctx context.Context, client crclient.Cli
 	return nodes.Items
 }
 
+func WaitForNReadyNodesByNodePool(t *testing.T, ctx context.Context, client crclient.Client, n int32, platform hyperv1.PlatformType, nodePoolName string) []corev1.Node {
+	g := NewWithT(t)
+	start := time.Now()
+
+	// waitTimeout for nodes to become Ready
+	waitTimeout := 30 * time.Minute
+	switch platform {
+	case hyperv1.PowerVSPlatform:
+		waitTimeout = 60 * time.Minute
+	}
+
+	t.Logf("Waiting for nodes to become ready by NodePool. NodePool: %s Want: %v", nodePoolName, n)
+	nodes := &corev1.NodeList{}
+	nodesFromNodePool := []corev1.Node{}
+	readyNodeCount := 0
+	err := wait.PollImmediateWithContext(ctx, 5*time.Second, waitTimeout, func(ctx context.Context) (done bool, err error) {
+		err = client.List(ctx, nodes)
+		if err != nil {
+			return false, nil
+		}
+		if len(nodes.Items) == 0 {
```

Comment:
I think this is necessary, let me elaborate on that.

Sequence:

- In the Poll we will fail until we have more than 0 nodes, so that check on line 237 is preventing to go far from that line. 
- After that it range over the slice and ensure if that node it's ready or not, then when 1 or more nodes are ready, it will not get blocked on the range.
- In line 251, it validate that the `nodesReady` are the same as `n` which are the nodes desired to be ready, so it will failAndPoll until this condition happen.

Make sense?